### PR TITLE
Simpler proof of naturality square

### PR DESCRIPTION
**Pull request by PHart3**
1. This pull request reduces the complexity of the proof of the pre-composition naturality square, contained in the Map-Nat folder.

- The main point is that it now shows that the map induced by a map of spans equals the canonical map by showing that both are in the same fiber of the post-composition equivalence. 
- This results in fewer long computations than those involved in the original proof.

2. It also does a good amount of clean-up and rearranging across the entire code base.
3. One notable change to the Dockerfile is installing happy-lib-2.0.2 instead of the latest version of Happy. It seems that the latter breaks the parser of Agda 2.6.3.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -19,7 +19,7 @@ RUN \
 RUN \
   mkdir -p /dist && \
   cabal update && \
-  cabal v2-install alex happy && \
+  cabal v1-install alex happy && \
```
Comment (favonia) [inline Dockerfile:22]: May I know why we are changing this back? I thought we changed it because `v1` had problems.

Comment (PHart3) [reply Dockerfile]: Your're right. I shouldn't have changed it back. Please disregard.

**Comment by favonia** [comment]
@PHart3 Thinking about it, if `v1-install` will work with a fixed version of `happy`, maybe it's better to also change `v2-install` back to `v1-install`.

**Comment by PHart3** [comment]
> @PHart3 Thinking about it, if `v1-install` will work with a fixed version of `happy`, maybe it's better to also change `v2-install` back to `v1-install`.

@favonia It does not work with happy-2.0.2. It generates a different error, an interface file error. See [here](https://github.com/PHart3/colimits-agda/actions/runs/11093171241/job/30819119528).

**Pull request merged**